src/components/StarRating/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactComponent as Star } from 'assets/svg/star.svg';
+import { useState } from 'react';
+import styles from './StarRating.module.scss';
+
+interface StarRatingProps {
+  rating: () => void;
+}
+
+function StarContainer({ rating }: StarRatingProps) {
+  const [hover, setHover] = useState(0);
+  const [click, setClick] = useState(0);
+  const clickHandler = (n: number) => {
+    setClick(n);
+    rating();
+  };
+
+  return (
+    <div className={styles.container}>
+      {[1, 2, 3, 4, 5].map((num) => (
+        <Star
+          fill={(hover >= num) || (click >= num) ? '#ff7f23' : '#eeeeee'}
+          key={num}
+          className={styles.container__star}
+          onMouseEnter={() => setHover(num)}
+          onMouseLeave={() => setHover(0)}
+          onClick={() => clickHandler(num)}
```

Comment:
`clickHandler` 라는 함수를 새롭게 만드신 이유가 궁금합니다.

Reply:
개인 차이는 있을 수 있겠지만 오히려 가독성을 해친다고 느껴집니다.
앞선 Props 를 수정한다면 다음과 같이 수정할 수 있을것 같아요

```suggestion
          onClick={() => {
            setClick(num);
            onClick?.();
          }}
```

Reply:
아 그렇군요..! 🥲 그럼 이번에 커스텀 훅`useRate`로 상태를 모두 옮겨주게 되었는데 이 경우에도 setClick과 onClick을 담는 함수를 반환해서 사용하는것이 아니라, setClick과 onClick를 모두 반환시켜 Star컴포넌트의 onClick함수를 구현하는 것이 가독성이 더 높다고 보시나요?

Reply:
훅 만드신걸 늦게 봤네요 😅 코멘트 남겼습니다 https://github.com/BCSDLab/JJBAKSA_FRONT_END/pull/22#discussion_r980240732